### PR TITLE
#93 Java8 ICsvTypedBeanWriter

### PR DESCRIPTION
**Pull request by longtimeago**
#93 Introducing new `ICsvTypedBeanWriter` (Java 8 ONLY) which is not relying on field names to map field to the column.
Instead of that it uses functional references.

See usage example in `org.supercsv.io.CsvTypedBeanWriterTest`

### CODE REVIEW

**Comment by ZioberMichal** [comment]
I checked your commit and it looks really good. @kbilsted, I think that we can merge it. Do you have any comment?

File: super-csv/src/test/java/org/supercsv/features/ReadingFeaturesTest.java [hunk 1]
```diff
@@ -316,7 +316,7 @@ public void testDateSupport() throws Exception {
 		Assert.assertEquals(new SimpleDateFormat("yyyy-MM-dd").parse("1999-07-12"), character.getBirthDate());
 	}
 	
-	@Test
```
Comment (ZioberMichal) [inline super-csv/src/test/java/org/supercsv/features/ReadingFeaturesTest.java]: Why you have commented this test?

**Comment by longtimeago** [comment]
@ZioberMichal was commented by mistake, enabled back, good catch!

**Comment by kbilsted** [comment]
@ziobermichal great, do merge.

**Comment by ZioberMichal** [comment]
Looks good now. Thanks for this PR. Next time when you do mistake with accident comment you can merge two commits to avoid having extra commit which does not do anything.

**Comment by longtimeago** [comment]
@ZioberMichal yep, sorry for "dirty" history. I'm still not very good in rebases ...

**Pull request merged**